ports/glutin/window.rs

```diff
@@ -671,44 +811,89 @@ fn create_window_proxy(_: &Window) -> Option<glutin::WindowProxy> {
-        TypedSize2D::new(width * scale_factor, height * scale_factor)
+        match self.kind {
+            WindowKind::Window(ref window) => {
+                let scale_factor = window.hidpi_factor() as u32;
+                // TODO(ajeffrey): can this fail?
+                let (width, height) = window.get_inner_size().expect("Failed to get window inner size.");
+                TypedSize2D::new(width * scale_factor, height * scale_factor)
+            }
+            WindowKind::Headless(ref context) => {
+                TypedSize2D::new(context.width, context.height)
+            }
+        }
     }
 
     fn size(&self) -> TypedSize2D<f32, ScreenPx> {
-        // TODO(ajeffrey): can this fail?
-        let (width, height) = self.window.get_inner_size().expect("Failed to get window inner size.");
-        TypedSize2D::new(width as f32, height as f32)
+        match self.kind {
+            WindowKind::Window(ref window) => {
+                // TODO(ajeffrey): can this fail?
+                let (width, height) = window.get_inner_size().expect("Failed to get window inner size.");
+                TypedSize2D::new(width as f32, height as f32)
+            }
+            WindowKind::Headless(ref context) => {
+                TypedSize2D::new(context.width as f32, context.height as f32)
+            }
+        }
     }
 
     fn client_window(&self) -> (Size2D<u32>, Point2D<i32>) {
-        // TODO(ajeffrey): can this fail?
-        let (width, height) = self.window.get_outer_size().expect("Failed to get window outer size.");
-        let size = Size2D::new(width, height);
-        // TODO(ajeffrey): can this fail?
-        let (x, y) = self.window.get_position().expect("Failed to get window position.");
-        let origin = Point2D::new(x as i32, y as i32);
-        (size, origin)
+        match self.kind {
+            WindowKind::Window(ref window) => {
+                // TODO(ajeffrey): can this fail?
+                let (width, height) = window.get_outer_size().expect("Failed to get window outer size.");
+                let size = Size2D::new(width, height);
+                // TODO(ajeffrey): can this fail?
+                let (x, y) = window.get_position().expect("Failed to get window position.");
+                let origin = Point2D::new(x as i32, y as i32);
+                (size, origin)
+            }
+            WindowKind::Headless(ref context) => {
+                let size = TypedSize2D::new(context.width, context.height);
+                (size, Point2D::zero())
+            }
+        }
+
     }
 
     fn set_inner_size(&self, size: Size2D<u32>) {
-        self.window.set_inner_size(size.width as u32, size.height as u32)
+        match self.kind {
+            WindowKind::Window(ref window) => {
+                window.set_inner_size(size.width as u32, size.height as u32)
+            }
+            WindowKind::Headless(..) => {}
```

Comment:
I think we can simplify this in these cases returning early and adding a `window()` method, or something, but I guess this is fine. r=me on the glutin changes.